tabular/src/autogluon/tabular/learner/abstract_learner.py

```diff
@@ -232,10 +258,16 @@ def score_debug(self, X: DataFrame, y=None, extra_info=False, compute_oracle=Fal
         for model_name, y_pred_proba_internal in model_pred_proba_dict.items():
             if self.eval_metric.needs_pred:
                 y_pred = self.label_cleaner.inverse_transform_proba(y_pred_proba_internal, as_pred=True)
-                scores[model_name] = self.eval_metric(y, y_pred)
+                y_tmp = y
+            else:
+                y_pred = self.label_cleaner.inverse_transform_proba(y_pred_proba_internal, as_pred=False)
+                y_tmp = y_internal
+            if not self.weight_evaluation:
+                scores[model_name] = self.eval_metric(y_tmp, y_pred)
             else:
-                y_pred_proba = self.label_cleaner.inverse_transform_proba(y_pred_proba_internal, as_pred=False)
-                scores[model_name] = self.eval_metric(y_internal, y_pred_proba)
+                if sample_weights is None:
+                    raise ValueError("Sample weights cannot be None when weight_evaluation is specified.")
+                scores[model_name] = compute_weighted_metric(y_tmp, y_pred, self.eval_metric, sample_weights)
```

Comment:
probably cleaner if this function is modified to also work if sample_weight is None, so we can simplify most of this code block to just a single call to

```
compute_weighted_metric(y_tmp, y_pred, self.eval_metric, sample_weights, self.weight_evaluation)
```

That would be a drop-in replacement for lines 265-270